### PR TITLE
Added filesystem and mount functions on disks

### PR DESCRIPTION
**Pull request by narasimhan-v**
avocado/utils/disk.py:
* Added creating filesystem, mount and unmount functionalities
* Fixed docstring

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
This module can be tested by [disk_fs_py.txt](https://github.com/avocado-framework/avocado/files/758683/disk_fs_py.txt) and output is [job_log.txt](https://github.com/avocado-framework/avocado/files/758685/job_log.txt)

File: avocado/utils/disk.py [hunk 1]
```diff
@@ -9,18 +9,84 @@
...
 import os
+from . import process
+
+
+LOGGER = logging.getLogger('avocado.test')
+
+
+class FSException(Exception):
+    """
+    Base Exception Class for all exceptions
```
Comment (ldoktor) [inline avocado/utils/disk.py]: As you have two of `for all exceptions` it implies it's not really for all exceptions. I assume you want to use one for `all filesystem-related exceptions` and the other for `disk-related exceptions`, right?

File: avocado/utils/disk.py [hunk 2]
```diff
@@ -9,18 +9,84 @@
 #
 # See LICENSE for more details.
 #
-# This code was inspired in the autotest project,
+# Copyright: 2017 IBM
+# Author: Narasimhan V sim@linux.vnet.ibm.com
 #
+# freespace() function was inspired in the autotest project,
```
Comment (ldoktor) [inline avocado/utils/disk.py]: well I'd say based on, but it's just wording and it is understandable...

File: avocado/utils/disk.py [hunk 3]
```diff
@@ -9,18 +9,84 @@
...
 
 def freespace(path):
+    """
+    Gets freespace on the given path.
+    """
     fs_stats = os.statvfs(path)
     return fs_stats.f_bsize * fs_stats.f_bavail
+
+
+def zero_disk(disk_name):
```
Comment (ldoktor) [inline avocado/utils/disk.py]: This is quite low-level function. I think it'd make sense to simply create `dd` method which would accept most of the `dd` arguments to allow generic usage. The defaults could be to override the file with zeros (I'm wondering why only 4096 bytes and not whole)

Comment (lmr) [inline avocado/utils/disk.py]: Using both block size and count=1 seems a mistake made by the author indeed.

File: avocado/utils/disk.py (hunk 1 above)
Comment (lmr) [inline avocado/utils/disk.py]: Yep, I agree that the docstrings need to be corrected. Right now the code has a copy/paste mistake.

Comment (narasimhan-v) [inline avocado/utils/disk.py]: yes, right

**Comment by narasimhan-v** [comment]
@ldoktor My bad. Did not check that partition covers these functionalities. Will look into this.
Thanks

File: avocado/utils/disk.py (hunk 3 above)
Comment (narasimhan-v) [inline avocado/utils/disk.py]: Not really. I wanted mainly to clear the FS headers, and thought of this way to do it.
And planned to extend functionalities in future.

**Comment by ldoktor** [comment]
@lmr the beauty is that `avocado.utils.partition` is already tested via a file-like device. So just by using this API from that module we'd get test coverage 😉 (also we have `avocado.utils.lv_utils` which shares some parts and could reuse some of the low level functions here and it is also tested by a functional test)

**Comment by narasimhan-v** [comment]
Closing this PR as the functionality needed are already there.

**Pull request closed without merging**